supabase/migrations/20260402000000_soft_delete_transactions.sql

```
@@ -0,0 +1,10 @@
+-- Add deleted_at column and indexes to expenses and incomes tables for soft delete support
+
+ALTER TABLE public.expenses ADD COLUMN IF NOT EXISTS deleted_at TIMESTAMPTZ;
+CREATE INDEX IF NOT EXISTS idx_expenses_deleted_at ON public.expenses (deleted_at) WHERE deleted_at IS NULL;
```

SQLFluff (4.2.2)

[error] CREATE INDEX should use CONCURRENTLY to avoid locking the table during the build.

(PG01)